### PR TITLE
Initial check in for ACS

### PR DESCRIPTION
**Pull request by alvadb**

### CODE REVIEW

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/CSAgentPoolProfile.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+         *
+         * @param <ParentT> the return type of the final {@link WithAttach#attach()}
+         */
+        interface WithVmSize<ParentT> {
+            /**
+             * Specify the size of the agents VMs.
+             * @param vmSize the size of the VM.
+             * @return the next stage of the definition
+             */
+            WithDnsLabel<ParentT> withVmSize(ContainerServiceVMSizeTypes vmSize);
```
Comment (jianghaolu) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/CSAgentPoolProfile.java]: VM?

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/ContainerService.java [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+@Beta
+public interface ContainerService extends
+        GroupableResource<ComputeManager, ContainerServiceInner>,
+        Refreshable<ContainerService>,
+        Updatable<ContainerService.Update> {
+
+    /**
+     * @return the properties of the orchestrator.
+     */
+    ContainerServiceOrchestratorProfile orchestratorProfile();
```
Comment (jianghaolu) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/ContainerService.java]: Do we need everything to be prefixed with `ContainerService`?

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ContainerServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    }
+
+    private ContainerServiceImpl withDiagnosticsProfile(boolean enabled) {
+        if (this.inner().diagnosticsProfile() == null) {
+            this.inner().withDiagnosticsProfile(new ContainerServiceDiagnosticsProfile());
+            this.inner().diagnosticsProfile().withVmDiagnostics(new ContainerServiceVMDiagnostics());
+
+        }
+
+        this.inner().diagnosticsProfile().vmDiagnostics().withEnabled(enabled);
```
Comment (jianghaolu) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ContainerServiceImpl.java]: could `vmDiagnostics()` be null?

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ContainerServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        return this;
+    }
+
+    @Override
+    public Update withAgentPoolCount(int agentPoolCount) {
+        if (agentPoolCount < 0 || agentPoolCount > 100) {
+            throw new RuntimeException("Agent pool count  must be in the range of 1 to 100 (inclusive)");
+        }
+
+        this.inner().agentPoolProfiles().get(0).withCount(agentPoolCount);
```
Comment (jianghaolu) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ContainerServiceImpl.java]: Will there always be at least one agent pool profile?

Comment (alvadb) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ContainerServiceImpl.java]: Yes

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/ContainerService.java (hunk 2 above)
Comment (alvadb) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/ContainerService.java]: This is generated class.

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/CSAgentPoolProfile.java (hunk 1 above)
Comment (alvadb) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/CSAgentPoolProfile.java]: fixed

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ContainerServiceImpl.java (hunk 3 above)
Comment (alvadb) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ContainerServiceImpl.java]: fixed

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/ContainerService.java [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+@Beta
+public interface ContainerService extends
+        GroupableResource<ComputeManager, ContainerServiceInner>,
+        Refreshable<ContainerService>,
+        Updatable<ContainerService.Update> {
+
+    /**
+     * @return the properties of the orchestrator
+     */
+    ContainerServiceOrchestratorProfile orchestratorProfile();
```
Comment (martinsawicki) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/ContainerService.java]: note for the future (not required for the initial PR) --- we'll probably want to flatten at least some of these "*Profile" thingies for better usability/discoverability of what matters.

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ComputeManager.java [hunk 6]
```diff
@@ -124,6 +126,16 @@ public AvailabilitySets availabilitySets() {
     }
 
     /**
+     * @return the availability set resource management API entry point
+     */
+    public ContainerServices containerServices() {
```
Comment (martinsawicki) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ComputeManager.java]: Missing `@Beta`

File: azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/Registry.java [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+
+        /**
+         * The stage of the registry definition allowing to enable admin user.
+         */
+        interface WithAdminUserEnabled {
+            /**
+             * Enable admin user.
+             * @return the next stage of the definition
+             */
+            Definition withAdminUserEnabled();
```
Comment (martinsawicki) [inline azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/Registry.java]: why does this return `Definition`?  Was `WithCreate` intended?

File: azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/Registry.java [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+
+            /**
+             * Disable admin user.
+             * @return the next stage of the definition
+             */
+            Definition withoutAdminUserEnabled();
```
Comment (martinsawicki) [inline azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/Registry.java]: same question as above

File: azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/implementation/ContainerRegistryManager.java [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+import com.microsoft.azure.management.containerregistry.Registries;
+import com.microsoft.azure.management.resources.fluentcore.arm.AzureConfigurable;
+import com.microsoft.azure.management.resources.fluentcore.arm.implementation.AzureConfigurableImpl;
+import com.microsoft.azure.management.resources.fluentcore.arm.implementation.Manager;
+import com.microsoft.rest.RestClient;
+
+/**
+ * Entry point to Azure compute resource management.
+ */
+public final class ContainerRegistryManager extends Manager<ContainerRegistryManager, ContainerRegistryManagementClientImpl> {
```
Comment (martinsawicki) [inline azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/implementation/ContainerRegistryManager.java]: missing `@Beta`

File: azure/src/main/java/com/microsoft/azure/management/Azure.java [hunk 10]
```diff
@@ -583,4 +589,18 @@ public SqlServers sqlServers() {
     public ServiceBusNamespaces serviceBusNamespaces() {
         return serviceBusManager.namespaces();
     }
+
+    /**
+     * @return entry point to managing Container Services.
+     */
+    public ContainerServices containerServices() {
```
Comment (martinsawicki) [inline azure/src/main/java/com/microsoft/azure/management/Azure.java]: Missing `@Beta`

File: azure/src/main/java/com/microsoft/azure/management/Azure.java [hunk 11]
```diff
@@ -583,4 +589,18 @@ public SqlServers sqlServers() {
...
+        return computeManager.containerServices();
+    }
+
+    /**
+     * @return entry point to managing Container Regsitries.
+     */
+    public Registries containerRegistries() {
```
Comment (martinsawicki) [inline azure/src/main/java/com/microsoft/azure/management/Azure.java]: MIssing `@Beta`

File: azure/src/test/java/com/microsoft/azure/management/TestContainerService.java [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+                .defineAgentPool("agentPool0" + newName)
+                .withCount(1)
+                .withVMSize(ContainerServiceVMSizeTypes.STANDARD_A1)
+                .withDnsLabel("ap0" + dnsPrefix)
+                .attach()
+                .withDcosOrchestration()
+                .withDiagnostics()
+                .withTag("tag1", "value1")
+                .create();
+        Assert.assertTrue("Container service not found.", resource.id() != null);
```
Comment (martinsawicki) [inline azure/src/test/java/com/microsoft/azure/management/TestContainerService.java]: for the future: there is Assert.assertNull(), .assertEquals(), etc.

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ComputeManager.java (hunk 6 above)
Comment (alvadb) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/ComputeManager.java]: Done

File: azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/Registry.java (hunk 7 above)
Comment (alvadb) [inline azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/Registry.java]: yes, good catch

File: azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/Registry.java (hunk 8 above)
Comment (alvadb) [inline azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/Registry.java]: yes, good catch

File: azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/implementation/ContainerRegistryManager.java (hunk 9 above)
Comment (alvadb) [inline azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/implementation/ContainerRegistryManager.java]: done

File: azure/src/main/java/com/microsoft/azure/management/Azure.java (hunk 10 above)
Comment (alvadb) [inline azure/src/main/java/com/microsoft/azure/management/Azure.java]: done

File: azure/src/main/java/com/microsoft/azure/management/Azure.java (hunk 11 above)
Comment (alvadb) [inline azure/src/main/java/com/microsoft/azure/management/Azure.java]: done

File: azure/src/test/java/com/microsoft/azure/management/TestContainerService.java (hunk 12 above)
Comment (alvadb) [inline azure/src/test/java/com/microsoft/azure/management/TestContainerService.java]: ok, makes sense

**Pull request merged**